src/Gjallarhorn/State.fs

```diff
@@ -82,3 +83,34 @@ type State<'TModel,'TMsg> (initialState : 'TModel, update : 'TMsg -> 'TModel ->
         member this.Dispose() = 
             (signal :?> System.IDisposable).Dispose()            
             GC.SuppressFinalize this
+
+/// A thread-safe wrapper for a reference value with change notification
+type AtomicMutable<'a when 'a : not struct>(value : 'a) as self =
+    let v = ref value
+    let deps =
+        let depsArray : ITracksDependents array = Array.empty
+        Dependencies.create depsArray self
+
+    /// Gets and sets the Value contained within this mutable
+    member this.Value 
+        with get() = !v
+        and set(value) =
```

Comment:
Note that we will need the compare exchange logic if you want to implement a Swap functionality...